src/libs/ajax/Metrics.ts

```diff
@@ -6,26 +6,27 @@ import { getConfig } from 'src/libs/config';
 import { withErrorIgnoring } from 'src/libs/error';
 import { MetricsEventName } from 'src/libs/events';
 import * as Nav from 'src/libs/nav';
-import { AuthState, authStore, getSessionId, getTerraUser } from 'src/libs/state';
+import { authStore, getSessionId, getTerraUser, MetricState, metricStore } from 'src/libs/state';
 import { v4 as uuid } from 'uuid';
 
 export const Metrics = (signal?: AbortSignal) => {
   const captureEventFn = async (event: MetricsEventName, details: Record<string, any> = {}): Promise<void> => {
     await ensureAuthSettled();
-    const state: AuthState = authStore.get();
-    const isRegistered = state.signInStatus === 'userLoaded';
+    const metricState: MetricState = metricStore.get();
+    const { signInStatus } = authStore.get();
+    const isRegistered = signInStatus === 'userLoaded';
     // If a user has not logged in, or has logged in but has not registered, ensure an anonymous ID has been generated.
     // The anonymous ID is used to associate events triggered by the anonymous user.
     // If the anonymous user registers during this session, the anonymous ID will be linked to their actual user ID.
-    if (!isRegistered && state.anonymousId === undefined) {
-      authStore.update((oldState: AuthState) => ({
+    if (!isRegistered && metricState.anonymousId === undefined) {
+      metricStore.update((oldState: MetricState) => ({
         ...oldState,
         anonymousId: uuid(),
       }));
     }
     const { buildTimestamp, gitRevision, terraDeploymentEnv } = getConfig();
     const signedInProps =
-      state.signInStatus === 'authenticated'
+      isRegistered || signInStatus === 'authenticated'
```

Comment:
also curious how this will change behavior

Reply:
this should theoretically not change behavior, `authenticated and` `userLoaded` are the two states that ensure that we have an auth token, and therefore an `idp` so it is just making sure that it will keep sending the `idp` in either state.